sql/src/main/java/org/opensearch/sql/sql/antlr/SQLSyntaxParser.java

```diff
@@ -32,4 +32,9 @@ public ParseTree parse(String query) {
     return parser.root();
   }
 
+  public boolean containsHints(String query) {
+    OpenSearchSQLLexer lexer = new OpenSearchSQLLexer(new CaseInsensitiveCharStream(query));
```

Comment:
Maybe: cache the `lexer` for performance? To avoid parsing entire query twice.

Reply:
ANTLR does not "play well" with lexer and parser re-use.

See this [thread](https://groups.google.com/g/antlr-discussion/c/B2TaUFm29jE) for context.

Reply:
Nice catch! We have to set [`REUSE_LEXER`](https://github.com/search?q=repo%3Aantlr/antlr4%20REUSE_LEXER&type=code) somehow or re-create the lexer for each query (re-parse).

Reply:
Re-creating the lexer is what co-author of ANTLR recommends.